### PR TITLE
feat(automation): clean and optimize

### PR DESCRIPTION
**Pull request by InteractiveTimmy**
# Description

The scope of the changes in this pull request are to update the automation pipeline to support a more flexible approach. The changes within this PR should allow for parallel runs over multiple platforms and packages.

### CODE REVIEW

File: scripts/packages.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  });
+});
+
+/**
+ * Get a list of all packages that depend on the provided target packages.
+ *
+ * @param {Array<string>} targetPackages Packages to find dependant packages of.
+ * @returns {Array<string>} Packages that depend on the provided target packages.
+ */
+const getDependantPackages = (targetPackages) => {
```
Comment (chrisadubois) [inline scripts/packages.js:83]: dependent (spelling error)

File: scripts/packages.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  }).parseSync();
+
+  const promises = [];
+
+  if (all) {
+    promises.push(getAllPackages());
+  }
+
+  if (changed) {
+    if (dependant) {
```
Comment (chrisadubois) [inline scripts/packages.js:123]: same

**Review by chrisadubois — APPROVED** [review]
small spelling, else looks great

**Comment by InteractiveTimmy** [comment]
> small spelling, else looks great

Fixed spelling.

**Pull request merged**